### ISSUE TITLE
Move away from api-observable API interface

### ISSUE THREAD

**Issue by jacogr** [polkadot-js/apps#507]
Replace all use of api-observable inside the application to use `@polkadot/api/promise`. 

Ok, doing a braindump here of a possible approach to make things less hairy and enable to piece-meal replacement. First things first, how does the API get injected?

- The API gets instantiated via https://github.com/polkadot-js/apps/blob/next/packages/ui-react-rx/src/Api/index.tsx
- In component use is done via the `withApi` HOC, i.e. https://github.com/polkadot-js/apps/blob/next/packages/ui-react-rx/src/with/api.tsx

So the hammer approach is to just replace ApiRx inside the first file... and then have everything break. A saner approach would be to instantiate the promise API alongside using the same provider (first file above). Once available, it can be injected using `withApi` which only adds the properties in the context.

Then we get down to use. Everywhere the API is used, we generally use a `withObservable` from https://github.com/polkadot-js/apps/blob/next/packages/ui-react-rx/src/with/observable.tsx

This one is simple, it just wraps the observable and gets the values of it into the props of the wrapping component. This allows us to alongside it, add a `withSubscription` (not `withPromise`, since we will generally want streaming values). This one can then use the api injected to make calls.

So far, so good. Then nice thing we have currently in api-observable is that is does a lot of combinations, i.e. it takes multiple subscriptions and makes it into a single value. This is used everywhere. So apart from having the subscriptions with single values, we would need a solution for this one...

(Ideas on that in a next comment...)

**Comment by jacogr** [comment]
Baby steps... (Typing this as I think through)

Ok, so after my braindump above, we have the API in a context and we have a way to inject the actual subscriptions. (Which is quite a bit of work in itself). So now getting down to some component to make sure it actually works.

A good place to start is the components inside ui-react-rx itself, i.e. a generic component that shows thecurrent chain, https://github.com/polkadot-js/apps/blob/next/packages/ui-react-rx/src/Chain.ts (just using this as a real-world example - and since it appears on the main screen at the right, we can see if it works or now)

So here we have an `observableDiv` HOC, basically just a `withObservable` that gets tied to a div. So in the use we want some way to specify the actual call we want to make and then the incoming value will just be formatted and displayed.

Here I'm suggesting that we take `chain` (a function on `api-observable`) and just point directly to what the actual promise api exposes, i.e. `rpc.system.chain` (so we would for others, here point to either RPC or queries, i.e. we may have `query.system,accountNonce` to retrieve a nonce.

Unlike api-observable, we don't need to re-invent the wheel here, we can just inject directly from the API exposed values. (Obviously in the case of combinations, it becomes a bit more involved.)

So basically, the subscription HOC will -

- parse the string into the components (rpc, system, chain, something like `const [area, section, fn] = param.split('.')`)
- make sure the endpoint actually exists, i.e. `assert(api[area] && api[area][section] && api[area][section[method], 'Cannot find api.${area}.${section}.${method}')`
- Make a subscription on this `const subId = await api[area][section][method]((value) => {... store value in state here to pass to HOC props... })`
- Obviously on unmount, we will unsubscribe
